include/odp/api/spec/pool_types.h

```diff
@@ -147,6 +147,38 @@ typedef struct odp_pool_stats_t {
 
 } odp_pool_stats_t;
 
+/**
+ * Pool statistics counters
+ *
+ * Same as odp_pool_stats_t excluding per thread counters.
+ */
+typedef struct odp_pool_stats_selected_t {
+	/** See odp_pool_stats_t::available */
+	uint64_t available;
+
+	/** See odp_pool_stats_t::alloc_ops */
+	uint64_t alloc_ops;
+
+	/** See odp_pool_stats_t::alloc_fails */
+	uint64_t alloc_fails;
+
+	/** See odp_pool_stats_t::free_ops */
+	uint64_t free_ops;
+
+	/** See odp_pool_stats_t::total_ops */
+	uint64_t total_ops;
+
+	/** See odp_pool_stats_t::cache_available */
+	uint64_t cache_available;
+
+	/** See odp_pool_stats_t::cache_alloc_ops */
+	uint64_t cache_alloc_ops;
+
+	/** See odp_pool_stats_t::cache_free_ops */
+	uint64_t cache_free_ops;
+
+} odp_pool_stats_selected_t;
```

Comment:
Why does this needs to be a separate structure? Cant we use odp_pool_stats_t itself ? 

What if thread_cache_available bit is set in odp_pool_stats_opt_t ? Where do we fill that information ?

Reply:
> Why does this needs to be a separate structure? Cant we use odp_pool_stats_t itself ?

The point of this function is to have minimal overhead, so `odp_pool_stats_selected_t` was added because `odp_pool_stats_t` is rather large due to the per thread counters.

> What if thread_cache_available bit is set in odp_pool_stats_opt_t ? Where do we fill that information ?

Good point, I'll add a sentence to the spec, which says that only fields available in `odp_pool_stats_selected_t` can be read.

Reply:
`odp_pool_stats_selected ()` spec updated in V7.

Reply:
Acked-by: Ashwin Sekhar T K <asekhar@marvell.com>